lua/lspconfig/angularls.lua

```diff
@@ -33,6 +33,15 @@ configs[server_name] = {
   on_new_config = function(new_config, new_root_dir)
     local new_probe_dir = get_probe_dir(new_root_dir)
 
+    -- Checking if we should provide legacy support
+    local file = io.open(new_probe_dir..'/@angular/core/package.json', 'r')
+    if file then
+        local file_contents = file:read('*all')
+        if tonumber(file_contents.match(file_contents, [["version": "(%d+).%d+.%d+"]])) < 9 then
+            table.insert(new_config.cmd, "--viewEngine")
```

Comment:
How is this not overwritten by https://github.com/neovim/nvim-lspconfig/blob/6062ab4a5ee0a2b0ccc1fe9531f8ee713114ea7c/lua/lspconfig/angularls.lua#L46-L53

Reply:
I couldn't get it to work the other way around, but I just saw what I got wrong.

Reply:
Well, actually I'm missing something either Lua level or at the language server level. In the README of the vscode implementation of the client they say that the backend is always shipped with the latest version, but the way we do here is by probing the project the user is currently working on. I guess that our decision is to make this a fallback.

In my case, I'm using Angular 6 and couldn't get to work with a "compatible" version of the language server since it asked for a minimum version 4.2 of typescript, which is not compatible with legacy projects.

If the user choose to use a custom cmd, providing a custom probing directory, my code does the trick, without breaking the usage with the default cmd. I'm not a Lua programmer, so I don't know exactly whats going on.

Should we just add this to the CONFIG.md and let the user copy the code if needed? I thought it would be nice to avoid the boilerplate.

Reply:
Let's move view_engine to a local variable and construct the table at the end.